main.go

```diff
@@ -0,0 +1,41 @@
+package main
+
+import (
+	"flag"
+	"log"
+	"net"
+	"os"
+
+	"google.golang.org/grpc"
+
+	changelogs "github.com/crossplane/crossplane-runtime/apis/changelogs/proto/v1alpha1"
+
+	"github.com/crossplane/changelogs-sidecar/server"
+)
+
+func main() {
+	// get socket file path, allowing user to specify via command line flag
+	socketPath := flag.String("socket-path", "/var/run/changelogs/changelogs.sock", "Path to create a Unix domain socket for change logs gRPC")
+	flag.Parse()
+
+	// clean up any existing socket file from previous runs
+	if err := os.Remove(*socketPath); err != nil && !os.IsNotExist(err) {
```

Comment:
I think of you use `errors.Is(err, fs.ErrNotExist)` it'll automatically handle the case where the error is nil. It's also what the `os.IsNotExist` godoc says new code should do (not sure why it's not marked deprecated though).

https://pkg.go.dev/os#IsNotExist

Reply:
oh interesting, happy to switch over to the recommended `errors.Is(err, fs.ErrNotExist)`, but I think we still need the `err != nil` check because we're negating the `errors.Is()` check here. we only want to enter the if block is there is an error and that error is something besides a "file not exist" error.

Without the nil check, I think we'd enter the if block when there is no error at all, which is not the desired behavior.
```
if err := os.Remove(*socketPath); !errors.Is(err, fs.ErrNotExist) {
  // nil error would enter this fatal block because it isn't an instance of fs.ErrNotExist
  log.Fatalf("...")
}
```